common/dbus.hpp

```diff
@@ -184,6 +187,34 @@ class DBusObject {
 	optional::optional<DBusMethodHandler<ReturnType>> GetMethodHandler(const MethodSpec &spec);
 };
 
+using DBusObjectPtr = shared_ptr<DBusObject>;
+
+class DBusServer : public DBusPeer {
+public:
+	DBusServer(events::EventLoop &loop, const string &service_name) :
+		DBusPeer(loop),
+		service_name_ {service_name} {};
+
+	~DBusServer() override;
+
+	error::Error AdvertiseObject(DBusObjectPtr obj);
+
+	// Only a convenience version for tests. Double-check that obj outlives the
+	// DBusServer!
+	error::Error AdvertiseObject(DBusObject &obj) {
```

Comment:
Since it is an overload, perhaps it should have "Test" in its name to prevent accidental usage?

Reply:
Not a strong opinion here, but it's just a variant for use with a reference, just like we have in many of our I/O classes. I just added the comment because I got myself caught in the trap of declaring `server` before `object` in the test.

Reply:
Ok, as long as there's no additional risk, it's fine.